code/lib/react-dom-shim/package.json

```diff
@@ -0,0 +1,74 @@
+{
+  "name": "@storybook/react-dom-shim",
+  "version": "7.0.0-beta.53",
+  "description": "",
+  "keywords": [
+    "storybook"
+  ],
+  "homepage": "https://github.com/storybookjs/storybook/tree/main/lib/react-dom-shim",
+  "bugs": {
+    "url": "https://github.com/storybookjs/storybook/issues"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/storybookjs/storybook.git",
+    "directory": "lib/react-dom-shim"
+  },
+  "funding": {
+    "type": "opencollective",
+    "url": "https://opencollective.com/storybook"
+  },
+  "license": "MIT",
+  "sideEffects": false,
+  "exports": {
+    ".": {
+      "node": "./dist/index.js",
+      "require": "./dist/index.js",
+      "import": "./dist/index.mjs",
+      "types": "./dist/index.d.ts"
+    },
+    "./react-18": {
```

Comment:
In order to support tools that do not understand `exports` (parcel, browserify, TS with `node` moduleResolution...), I think we need an actual file at the root of the project named `react-18.js` as well.  That's the reason there's a `/preset.js` that re-exports from dist.

Reply:
Hmm, well none of those are going to know to put the alias in place? Or are you imagining a future parcel builder or similar?

Reply:
I changed this, see rule above https://github.com/storybookjs/storybook/pull/21197#discussion_r1115472786